main/src/ca/uwaterloo/flix/api/lsp/Indexer.scala

```diff
@@ -207,6 +207,9 @@ object Indexer {
       val tpe = Type.mkRegion(sym.tvar, loc)
       Index.occurrenceOf(sym, tpe) ++ visitExp(exp) ++ Index.occurrenceOf(exp0)
 
+    case Expression.OnExit(exp1, exp2, _, _, _, _) =>
```

Comment:
I would like to rename `OnExit`  to `ScopeExit` everywhere. 

This name is probably not that much better, but it reflects the name of a very similar construct we use in our paper + Isabelle proof, so to keep things consistent.